### ISSUE TITLE
Regression (false positive) with require - Unable to resolve path to module (import/no-unresolved)

### ISSUE THREAD

**Issue by alasdairhurst** [import-js/eslint-plugin-import#2186]
We're using mingo@2.5.3 and noticed a regression with the no-unresolved rule

```
const Mingo = require('mingo');
```

Suddenly started resulting in the error:
```
Unable to resolve path to module 'mingo' import/no-unresolved
```

Not fully versed with how typescript imports work, but Mingo also uses it. My module does not.
https://github.com/kofrasa/mingo/tree/2.5.3

note we're also not using the "import" syntax here, but require.  Maybe plugin-import is mixing up "main" and "module" in package.json?

```
"main": "./dist/mingo.js",
 "module": "./lib/index.js",
```

the file at "module" is not available in this case.
![image](https://user-images.githubusercontent.com/8441810/128748028-9475cbec-9246-45c3-9678-f04e02034f21.png)

**Comment by kachkaev** [comment]
Seeing the same after upgrading `eslint-import-resolver-node` from `0.3.4` to `0.3.5`. Temp workaround in `package.json`:

```json
{
  "resolutions": {
    "**/eslint-import-resolver-node": "0.3.4"
  }
}
```

**Comment by plazek** [comment]
We are seeing the same behavior on the version with ES6 imports as well.

**Comment by ljharb** [comment]
@alasdairhurst mingo 2.5.3 indeed is broken; its "module" field points at a nonexistent file. Its "main", however, works, and this is indeed the major change in v0.3.5 of the node resolver.

While this is a correct error - the mingo maintainers would need to fix this (but probably won't, since they're up to v4 and this is a bug in v2) - it seems reasonable for the "module" field lookup, when the file does not exist, to fall back to checking the "main".

**Comment by ljharb** [comment]
@kachkaev @plazek if this isn't the same root cause for either of you - meaning, if the package that's being warned on actually has an *existing* file where the "module" field points - please do file a new issue with as many details as possible.
